packages/@okta/vuepress-site/docs/guides/unlock-mobile-app-with-biometrics/retrieve-refresh-token/android/getnewaccesstoken.md

```diff
@@ -1 +1,17 @@
-Example code: get a new access token
\ No newline at end of file
+Then override `onActivityResult` to get the results from the `Keyguard`
+
+```java
+@Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (requestCode == REQUEST_CODE_CREDENTIALS && resultCode == RESULT_OK) {
+            if (keyguardEncryptionManager.getCipher() == null) {
+                keyguardEncryptionManager.recreateCipher();
+            }
+        } else {
+            webAuth.handleActivityResult(requestCode, resultCode, data);
+        }
+    }
+```
+
+After `RESULT_OK` is returned the device is authenticated and `keyguardEncryptionManager` is able
+to decrypt the all the encrypted data including the access and refresh tokens.
```

Comment:
After `RESULT_OK` is returned, the device is authenticated and `keyguardEncryptionManager` is able
to decrypt all of the encrypted data including the access and refresh tokens.